services/wallet/transfer/transaction_manager_multitransaction.go

```diff
@@ -491,3 +491,67 @@ func (tm *TransactionManager) getVerifiedWalletAccount(address, password string)
 		AccountKey: key,
 	}, nil
 }
+
+func (tm *TransactionManager) removeMultiTransactionByAddress(address common.Address) error {
+	// We must not remove those transactions, where from_address and to_address are different and both are stored in accounts DB
+	// and one of them is equal to the address, as we want to keep the records for the other address
+	// That is why we don't use cascade delete here with references to transfers table, as we might have 2 records in multi_transactions
+	// for the same transaction, one for each address
+
+	stmt, err := tm.db.Prepare(`SELECT rowid, from_address, to_address
+								FROM multi_transactions
+								WHERE from_address=? OR to_address=?`)
+	if err != nil {
+		return err
+	}
+
+	rows, err := stmt.Query(address, address)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	rowIDs := make([]int, 0)
+	rowID, fromAddress, toAddress := 0, common.Address{}, common.Address{}
+	for rows.Next() {
+		err = rows.Scan(&rowID, &fromAddress, &toAddress)
+		if err != nil {
+			log.Error("Failed to scan row", "error", err)
+			continue
+		}
+
+		// Remove self transactions as well, leave only those where we have the counterparty in accounts DB
+		if fromAddress != toAddress {
+			// If both addresses are stored in accounts DB, we don't remove the record
+			var addressToCheck common.Address
+			if fromAddress == address {
+				addressToCheck = toAddress
+			} else {
+				addressToCheck = fromAddress
+			}
+			cntrpartyExists, err := tm.accountsDB.AddressExists(types.Address(addressToCheck))
```

Comment:
I feel that using `counterparty` instead of `cntrparty` is more readable, those few saved letters don't help readability in any way.